src/coreclr/src/gc/gc.cpp

```diff
@@ -32558,20 +32451,20 @@ size_t gc_heap::desired_new_allocation (dynamic_data* dd,
                     new_allocation = new_allocation1;
                 }
             }
-            else //large object heap
+            else // not a SOH generation
             {
                 uint32_t memory_load = 0;
                 uint64_t available_physical = 0;
                 get_memory_info (&memory_load, &available_physical);
 #ifdef TRACE_GC
                 if (heap_hard_limit)
                 {
-                    size_t loh_allocated = 0;
-                    size_t loh_committed = committed_size (true, &loh_allocated);
-                    dprintf (1, ("GC#%Id h%d, GMI: LOH budget, LOH commit %Id (obj %Id, frag %Id), total commit: %Id (recorded: %Id)",
-                        (size_t)settings.gc_index, heap_number,
-                        loh_committed, loh_allocated,
-                        dd_fragmentation (dynamic_data_of (max_generation + 1)),
+                    size_t uoh_allocated = 0;
+                    size_t uoh_committed = committed_size (gen_number, &uoh_allocated);
```

Comment:
this is not necessarily `loh` generation either. I will just call it `allocated` and `committed`. They are just numbers reported by the `dprintf`

---
In reply to: [367271669](https://github.com/dotnet/runtime/pull/1688#discussion_r367271669) [](ancestors = 367271669)

Reply:
sure they are not necessarily loh gen, but the point is for any uoh gen you will be returning the correct # for committed/allocated and that's not the case for gen0/1.

Reply:
It is now:
```c++
else // not a SOH generation
            {
                uint32_t memory_load = 0;
                uint64_t available_physical = 0;
                get_memory_info (&memory_load, &available_physical);
#ifdef TRACE_GC
                if (heap_hard_limit)
                {
                    size_t allocated = 0;
                    size_t committed = uoh_committed_size (gen_number, &allocated);
                    dprintf (1, ("GC#%Id h%d, GMI: UOH budget, UOH commit %Id (obj %Id, frag %Id), total commit: %Id (recorded: %Id)", 
                        (size_t)settings.gc_index, heap_number, 
                        committed, allocated,
                        dd_fragmentation (dynamic_data_of (gen_number)),
                        get_total_committed_size(), (current_total_committed - current_total_committed_bookkeeping)));
                }
#endif //TRACE_GC
. . .
```

what should it be?

---
In reply to: [368124892](https://github.com/dotnet/runtime/pull/1688#discussion_r368124892) [](ancestors = 368124892)

Reply:
not sure what you mean? you said "this is not necessarily loh generation either.", and I said that's true but it will return correct info for other uoh gen anyway. the updated code looks fine to me.

Reply:
I just meant the `loh`  should not be a part of the name either.

---
In reply to: [368133805](https://github.com/dotnet/runtime/pull/1688#discussion_r368133805) [](ancestors = 368133805)